### DISCUSSION TITLE
Adding npm Package from GitHub

### DISCUSSION THREAD

**Discussion by ben-laird** [denoland/deno#26566] [Q&A]
Hello!

I'm very excited for the release of Deno 2, and I've been looking forward to using the [ECMAScript Signals proposal](https://github.com/tc39/proposal-signals) package in Deno to get a feel for the proposal's concepts. Regrettably, I don't know if the [package itself](https://github.com/proposal-signals/signal-polyfill) is published to npm, so I'd like to reference it via GitHub.

My question is how does one add, install, or otherwise reference an npm package using a GitHub URL?

I know there are other issues that discuss something similar, namely [#5543](https://github.com/denoland/deno/issues/5543) and [#24977](https://github.com/denoland/deno/discussions/24977) (as well as this [issue about adding modules from branches](https://github.com/denoland/deno_registry2/issues/78)), but this is a bit different considering that Deno now has full Node & npm support.

Thanks in advance for any help you can provide!

**Comment by PaulRBerg** [comment]
I have the same question and I really need this. I can't use Deno without this feature.

With Bun and Node.js, I can install GitHub-based dependencies like so:

```json
"@sablier/devkit": "github:sablier-labs/devkit#main"
```

Where `#main` is the main branch.

**Comment by johngeorgewright** [comment]
I got around this using [esm.sh](https://esm.sh/#docs). For example:

```ts
import 'https://esm.sh/gh/proposal-signals/signal-polyfill'
```

**Reply by adaliszk** [reply]
What do you do when it says "error: Module not found"?
Though I ugess I did put it into the imports map, and not using it as strait import.
